### PR TITLE
test: add tests for retry function

### PR DESCRIPTION
**Pull request by mxz42**
Add two tests for the retry function. The first test should run the closure only once as it immediately succeeds. The closure of the second test always fails, hence the closure should be run `max_attempt` times.

### CODE REVIEW

File: gateway/ln-gateway/src/utils.rs [hunk 1]
```diff
@@ -43,3 +43,40 @@ where
...
+    use anyhow::anyhow;
+
+    use super::retry;
+
+    #[tokio::test]
+    async fn retry_one_attempt() {
+        let counter = AtomicU8::new(0);
+        let closure = || async {
+            counter.fetch_add(1, Ordering::SeqCst);
+            Ok(42)
```
Comment (m1sterc001guy) [inline gateway/ln-gateway/src/utils.rs:61]: Why 42?

Comment (dpc) [reply gateway/ln-gateway/src/utils.rs:61]: Because it's the best number?

Comment (mxz42) [reply gateway/ln-gateway/src/utils.rs:61]: > Because it's the best number?

yeah, it's literally that @m1sterc001guy  
no real reason at all ;D

Comment (mxz42) [reply gateway/ln-gateway/src/utils.rs:62]: done

**Review by dpc — APPROVED** [review]

**Pull request merged**